src/gauge/constructs/fat7LinkConstructs.h

```diff
@@ -8,227 +8,348 @@
 #include "../../base/math/gsu3.h"
 #include "../gaugefield.h"
 #include "../../modules/HISQ/smearParameters.h"
+#include <assert.h>
 
 
 template<class floatT, size_t HaloDepth, CompressionType comp>
-    __host__ __device__ GSU3<floatT> inline naikLinkStaple(gaugeAccessor<floatT,comp> gAcc, gSiteMu siteMu) {
+__host__ __device__ GSU3<floatT> inline naikLinkStaple(gaugeAccessor<floatT,comp> gAcc, gSiteMu siteMu) {
 
-    typedef GIndexer<All,HaloDepth> GInd;
+  typedef GIndexer<All,HaloDepth> GInd;
 
-    gSite origin = GInd::getSite(siteMu.isite);
-    int mu = siteMu.mu;
-    gSite down = GInd::site_dn(origin,mu);
-    gSite up = GInd::site_up(origin,mu);
+  gSite origin = GInd::getSite(siteMu.isite);
+  int mu = siteMu.mu;
+  gSite down = GInd::site_dn(origin,mu);
+  gSite up = GInd::site_up(origin,mu);
     
-    return gAcc.getLink(GInd::getSiteMu(down,mu))*gAcc.getLink(GInd::getSiteMu(origin,mu))*gAcc.getLink(GInd::getSiteMu(up,mu));
+  return gAcc.getLink(GInd::getSiteMu(down,mu))*gAcc.getLink(GInd::getSiteMu(origin,mu))*gAcc.getLink(GInd::getSiteMu(up,mu));
 }
 
 template<class floatT,size_t HaloDepth,CompressionType comp>
-  __host__ __device__ GSU3<floatT> inline threeLinkStaple(gaugeAccessor<floatT,comp> gAcc, gSiteMu siteMu) {
-    typedef GIndexer<All,HaloDepth> GInd;
-    GSU3<floatT> temp = gsu3_zero<floatT>();
-    gSite site = GInd::getSite(siteMu.isite);
-    gSite site_save = site;
-    int mu = siteMu.mu;
+__host__ __device__ GSU3<floatT> inline threeLinkStaple(gaugeAccessor<floatT,comp> gAcc, gSiteMu siteMu, int excluded_dir1 = -1, int excluded_dir2 = -1) {
```

Comment:
I think it would be good to put the hyp-smearing specific changes into a separate threeLinkStaple and call it hypThreeLinkStaple_first_level or something consistent with the other hyp functions in order to keep it separated from the HISQ functions.

Reply:
I agree, that makes sense.  I did not spend very long thinking about naming conventions for this port, so I am generally fine with whatever changes need to be made in that regard.